### PR TITLE
#353 Add the ability to edit meeting

### PR DESCRIPTION
**Pull request by Ymkavi**
https://user-images.githubusercontent.com/58224993/191234061-40e2c5ef-fb69-4dda-9520-679b91f1085d.mp4

### CODE REVIEW

**Review by YuraRiabov — APPROVED** [review]

File: frontend/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -36,6 +36,10 @@ const routes: Routes = [
         path: 'new-meeting',
         loadChildren: () => import('./modules/new-meeting/new-meeting.module').then((m) => m.NewMeetingModule),
     },
+    {
+        path: 'edit-meeting',
+        loadChildren: () => import('./modules/edit-meeting/edit-meeting.module').then((m) => m.EditMeetingModule),
+    },
```
Comment (tatianahutii) [inline frontend/src/app/app-routing.module.ts]: Why do we need a separate module for this?

File: frontend/src/app/modules/edit-meeting/edit-meeting/edit-meeting.component.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        this.activateRoute.params.pipe(this.untilThis).subscribe((params) => {
+            this.spinnerService.show();
+            this.http
+                .getMeetingById(params['id'])
+                .pipe(this.untilThis)
+                .subscribe((meeting) => {
+                    this.meeting = meeting;
+
+                    this.spinnerService.hide();
+                });
```
Comment (tatianahutii) [inline frontend/src/app/modules/edit-meeting/edit-meeting/edit-meeting.component.ts]: Please get rid of subscribe in subscribe

File: frontend/src/app/modules/new-meeting/new-meeting/new-meeting.component.ts [hunk 3]
```diff
@@ -302,6 +352,69 @@ export class NewMeetingComponent extends BaseComponent implements OnInit, OnDest
...
+
+        this.userService
+            .getCurrentUser()
+            .pipe(this.untilThis)
+            .subscribe((currentUser) => {
+                this.meetingToEdit.meetingMembers
+                    .filter((member) => member.id !== currentUser.id)
+                    .forEach((member) => this.addMemberToList(member));
+            });
+    }
```
Comment (tatianahutii) [inline frontend/src/app/modules/new-meeting/new-meeting/new-meeting.component.ts]: Move to a separate method

File: frontend/src/app/modules/new-meeting/new-meeting/new-meeting.component.ts [hunk 4]
```diff
@@ -302,6 +352,69 @@ export class NewMeetingComponent extends BaseComponent implements OnInit, OnDest
...
+        }
+
+        this.customTimeShown = true;
+
+        this.meetingForm.patchValue({
+            duration: this.duration,
+            customTime: this.duration.minutes,
+            unitOfTime: this.duration.unitOfTime,
+        });
+    }
```
Comment (tatianahutii) [inline frontend/src/app/modules/new-meeting/new-meeting/new-meeting.component.ts]: Need refactoring

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/MeetingService.cs [hunk 5]
```diff
@@ -24,6 +24,52 @@ public MeetingService(EasyMeetsCoreContext context, IMapper mapper, IUserService
...
+                .Include(m => m.ExternalAttendees)
+                .Include(m => m.MeetingMembers)
+                    .ThenInclude(meetingMember => meetingMember.TeamMember)
+                    .ThenInclude(teamMember => teamMember.User)
+                .FirstOrDefaultAsync(m => m.Id == id);
+
+            if (meeting is null)
+            {
+                throw new KeyNotFoundException($"Meeting with id {id} does not exist");
+            }
```
Comment (apofis96) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/MeetingService.cs]: combine with ??

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/MeetingService.cs [hunk 6]
```diff
@@ -24,6 +24,52 @@ public MeetingService(EasyMeetsCoreContext context, IMapper mapper, IUserService
...
+        }
+
+        public async Task<SaveMeetingDto> UpdateMeetingAsync(SaveUpdateMeetingDto updateMeetingDto)
+        {
+            var meeting = GetByIdInternal(updateMeetingDto.Id);
+
+            if (meeting is null)
+            {
+                throw new KeyNotFoundException($"The meeting with {updateMeetingDto.Id} id not found");
+            }
```
Comment (apofis96) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/MeetingService.cs]: combine with ??

**Review by apofis96 — CHANGES_REQUESTED** [review]

**Review by apofis96 — APPROVED** [review]

**Pull request merged**